lldb/source/Plugins/TypeSystem/Swift/TypeSystemSwiftTypeRef.cpp

```diff
@@ -3177,7 +3186,15 @@ TypeSystemSwiftTypeRef::GetTypedefedType(opaque_compiler_type_t type) {
     }
     return std::get<CompilerType>(pair);
   };
-
+#ifndef NDEBUG
+  // We skip validation when dealing with a builtin type since builtins are
+  // considered type aliases by Swift, which we're deviating from.
```

Comment:
I'm surprised that this is necessary. Wouldn't the typedefed type of a builtin return the same result in both typesystems (= the real builtin type)?

Reply:
Maybe I misunderstood what we decided to do. I thought we'd like to deviate from `SwiftASTContext` and not consider builtins to be typedefs, because otherwise that could cause infinite loops because the "resolved" `BuiltinTypeName` is just itself. Would this not be a problem?

Reply:
Oh! I get it now. I thought in the case that the input isn't a typealias, GetTypedefedType() would return the original type, but it returns an empty type instead. So I think this makes some sense.